src/syft/core/node/common/action/save_object_action.py

```diff
@@ -49,7 +49,10 @@ def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
                     if hasattr(self.obj, "description")
                     else ""
                 ),
-                read_permissions=set([verify_key, node.verify_key]),
+                read_permissions={
+                    node.verify_key: node.id,
+                    verify_key: None,  # we dont have the passed in sender's UID
```

Comment:
if we don't pass in the sender's verify_key then they won't be able to download the thing they sent. This is incorrect.

Reply:
Correction - I read this wrong. I was reading the key:value paring as a type hint. This makes sense.

Reply:
Yeah sorry, this is ugly, we are overloading the permissions with a separate concept which is a log of requests being accepted vs rejected. This was the quickest way to get check "accepted" working. I don't like it but introducing a whole separate thing or not providing check accept also seemed wrong right now.